server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -330,6 +350,14 @@ func (e *EventChannel) handleEvent(event *pepb.PublishBuildToolEventStreamReques
 		}
 	}
 
+	if len(e.eventsBeforeStarted) > 0 {
+		previousEvent := e.eventsBeforeStarted[0]
+		e.eventsBeforeStarted = e.eventsBeforeStarted[1:]
```

Comment:
Will this be a NPE if eventsBeforeStarted is 1?


(could we rewrite this a just a loop over e.eventsBeforeStarted?

Reply:
I initially just had a loop - but that leads to an infinite loop because this thing calls handleEvent for each of these prior events and it doesn't pop them off each time.

I have a test that eventually makes it down to 1 and then 0 eventsBeforeStarted and it doesn't NPE. Just tested this out in go playground and when you access [1:] on a slice with one element - it returns an empty slice. If you access it on a slice with 0 elements, you get an out of bounds error - but we check `len(e.eventsBeforeStarted) > 0` here.